pkg/types/baremetal/validation/platform.go

```diff
@@ -61,6 +65,57 @@ func validateOSImageURI(uri string) error {
 	return nil
 }
 
+// validateHosts checks that hosts have all required fields set with appropriate values
+func validateHosts(hosts []*baremetal.Host, fldPath *field.Path) field.ErrorList {
+	hostErrs := field.ErrorList{}
+
+	values := make(map[string]map[interface{}]struct{})
+
+	//Initialize a new validator and register a custom validation rule for the tag `uniqueField`
+	validate := validator.New()
+	validate.RegisterValidation("uniqueField", func(fl validator.FieldLevel) bool {
```

Comment:
I'd think this should probably go somewhere in pkg/validate, in an init() function, so it's generally usable by everyone

Reply:
We could wait for this one and move it out later if someone else other than baremetal would use it.